lib/Listener.js

```diff
@@ -69,6 +69,71 @@ class Listener {
     return this._started
   }
 
+  pause (cold) {
+    if (!this._started) {
+      return Promise.resolve(this)
+    }
+
+    if (this._paused) {
+      if (this._resuming) {
+        console.warn('Tried to pause listener whilst busy resuming')
+      }
+
+      return this._paused
+    }
+
+    this._paused = new Promise((resolve, reject) => {
+      const ops = [this._consumer.cancel(this._consumerTag)]
+
+      if (cold) {
+        // cold pause requested, so let's push all messages
+        // back in to the queue rather than handling them
+        this._cold = true
+        ops.push(this._consumer.recover())
+      }
+
+      return Promise.all(ops)
+        .then(() => resolve(this))
+        .catch(reject)
+    })
+
+    return this._paused
+  }
+
+  resume () {
+    if (this._resuming) return this._resuming
+    if (!this._started) return this.start()
+    if (!this._starting && !this._paused) return Promise.resolve(this)
+
+    this._resuming = new Promise(async (resolve, reject) => {
+      let consumeResult
+
+      try {
+        consumeResult = await this._consumer.consume(
+          this._options.queue,
+          bubble.bind(this._incoming.bind(this)),
```

Comment:
assuming this is the callback, should it not throw in _incoming() rather than silently ignore messages, for e.g. when data can not be parsed?

Reply:
You mean instead of `nack`ing at [lib/Listener.js:146](https://github.com/jpwilliams/remit/blob/8a87a57ae65001cc1871bae346cba0b28ed3c6a3/lib/Listener.js#L146)?

'tis probably out of scope for this particular PR as I've only moved that consumption block rather than changing it, but `nack`ing is probably our best bet.

If an invalid message comes in, throwing would mean killing the process, but `nack`ing means we can use dead letter queues in the future.

We could well throw a `console.warn` in there to help, or perhaps an emission that emits all messages that fail (via an EventEmitter; not remit)?